app/models/search/find_lot_by_batch_id.rb

```diff
@@ -1,7 +1,16 @@
 # frozen_string_literal: true
-class Search::FindLotByBatchId < Search # rubocop:todo Style/Documentation
+# Used by Gatekeeper to find the lot associated with a batch.
+# Not used nowadays, as QC is no longer performed in house.
+class Search::FindLotByBatchId < Search
   def scope(criteria)
-    root_asset = Batch.find_by(id: criteria['batch_id']).try(:parent_of_purpose, 'Tag PCR')
+    root_asset = parent_of_purpose(Batch.find_by(id: criteria['batch_id']))
     Lot.with_qc_asset(root_asset)
   end
+
+  def parent_of_purpose(batch)
```

Comment:
Not understanding what code climate is whinging about with these 'doesn't depend on instance state' messages

Reply:
So code climate doesn't like it because its looking at this and thinking:

'Why are you asking search to do this, it doesn't depend on anything that search knows, we're just asking questions of batch, shouldn't this method be on batch'

Usually this complaint ignores the fact that for a duck typed language like ruby, the class is part of the state. That said, I'm not sure this really applies here, and it could be argued that batch could have an 'ancestors_of_purpose' method, and this search just passes in 'Tag PCR'.

However, one of the other thing code climate gets incredibly sad about is Batch. And the number of methods in Batch. And this method is quite specific to this lookup here. So I don't think moving it to batch would make much sense.